src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperDescriptorJsonConverter.cs

```diff
@@ -906,5 +914,141 @@ private static void ReadToEndOfCurrentObject(JsonReader reader)
 
             throw new JsonSerializationException($"Could not read till end of object, end of stream. Got '{reader.TokenType}'.");
         }
+
+        private static string Intern(string str)
```

Comment:
Add a comment on why we're using our own weak reference table and not string.intern